src/bin/opensmt.cc

```diff
@@ -238,4 +210,83 @@ void catcher( int sig )
   }
 }
 
+
+void printHelp(int code = 0)
+{
+    const char help_string[]
+        = "Usage: ./opensmt [OPTION] <filename>\n"
+          "where OPTION can be\n"
+          "  --help [-h]                     prints this help message and exits\n"
+          "  --version                       prints opensmt version and exits\n"
+          "  --dry-run [-d]                  executes dry run\n"
+          "  --random-seed [-r] <seed>       sets random seed to specific number\n"
+          "  --produce-interpolants [-i]     enables interpolant computation\n"
+          "  --verbose [-v]                  verbose run of the solver\n"
+          "  --pipe [-p]                     for execution within a pipe\n";
+    std::cerr << help_string;
+    exit(code);
+}
+
+SMTConfig parseCMDLineArgs( int argc, char * argv[ ] )
+{
+    SMTConfig res;
+    int versionOpt = 0;
+    int dryRunOpt = 0;
+
+    struct option long_options[] =
+        {
+            {"help", no_argument, nullptr, 'h'},
+            {"version", no_argument, &versionOpt, 1},
+            {"verbose", no_argument, nullptr, 'v'},
+            {"dry-run", no_argument, &dryRunOpt, 'd'},
+            {"produce-interpolants", no_argument, nullptr, 'i'},
+            {"pipe", no_argument, nullptr, 'p'},
+            {"random-seed", required_argument, nullptr, 'r'},
+            {0, 0, 0, 0}
+        };
+
+    while (true) {
+        int option_index = 0;
+        int c = getopt_long(argc, argv, "r:hdivp", long_options, &option_index);
+        if (c == -1) { break; }
+
+        switch (c) {
+            case 0:
+                if (long_options[option_index].flag == &versionOpt) {
+                    std::cerr << OPENSMT_GIT_DESCRIPTION << "\n";
+                    exit(0);
+                } else {
+                    printHelp(1);
```

Comment:
I refer to your comment above ...

Reply:
This line is reached when user runs
```
./opensmt --dry-run
```

Reply:
I think it is caused by using `dryRunOpt` that should not be used since there also exists short version `-d`.
Hence I would remove `dryRunOpt` and replace this else branch with `assert(false)`.

Reply:
It was reached with --dry-run, but I've updated it, this code was not needed